### PR TITLE
Fix cross entropy patch for LLama

### PR DESCRIPTION
**Pull request by pramodith**
## Summary
This PR addresses issues #368 and partially addresses #369 by fixing the patch for Llama based models.

## Details
Some recent changes to HF's transformers lib meant that the training scripts for llama models no longer defined a `cross_entropy` attribute, rather they included a `loss_function` [attribute](https://github.com/huggingface/transformers/blob/a06a0d12636756352494b99b5b264ac9955bc735/src/transformers/models/llama/modeling_llama.py#L1224) that effectively mapped to torch's `nn.cross_entropy` [function](https://github.com/huggingface/transformers/blob/main/src/transformers/loss/loss_utils.py#L32).

These changes invalidated the way we patched up the models with LigerCrossEntropy.

In this PR we correct the patch in LLama by:

- Allowing `LigerCrossEntropyLoss` to accept kwargs in the constructor.
- Update the patched forward pass to also consider the implications of using `LigerCrossEntropyLoss`

_Note: Additional PR's might be needed to patch other models._
## Testing Done
Added a new test case in our convergence tests, to ensure that the patched cross entropy loss works.
- Hardware Type: A100 SXM 80GB
- [X] run `make test` to ensure correctness
- [X] run `make checkstyle` to ensure code style
- [X] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by ByronHsu** [comment]
This PR is covered by https://github.com/linkedin/Liger-Kernel/pull/375. Feel free to close it and open a new one on the main's branch for the follow-up works (e.g. rms, rope functional)
